pkg/runctl/controller.go

```diff
@@ -275,12 +275,8 @@ func (c *Controller) syncHandler(key string) error {
 		run, err := runManager.GetRun(pipelineRun)
 		if err != nil {
 			pipelineRun.StoreErrorAsMessage(err, "error syncing resource")
-			if err = c.changeState(pipelineRun, api.StateCleaning); err != nil {
-				return err
-			}
-			pipelineRun.UpdateResult(api.ResultErrorInfra)
-			c.metrics.CountResult(api.ResultErrorInfra)
-			return nil
+			// should we count errors here?
```

Comment:
Controllers typically emit events for the resources they work on in error conditions.